app/src/main/java/com/eveningoutpost/dexdrip/LibreAlarmReceiver.java

```diff
@@ -168,6 +171,20 @@ public void run() {
     }
 
     public static void processReadingDataTransferObject(ReadingData.TransferObject object) {
+    	Log.i(TAG, "Data that was recieved from librealarm is " + HexDump.dumpHexString(object.data.raw_data));
+    	// Save raw block record (we start from block 0)
+        LibreBlock.createAndSave("LibreAlarm", object.data.raw_data, 0);
+
+        if(Home.getPreferencesBooleanDefaultFalse("external_blukon_algorithm")) {
```

Comment:
Since you created this PR there has been a refactoring in xDrip master (moving the static preference handling to its own class) - this needs a change here:

Change `Home.getPreferencesBooleanDefaultFalse("external_blukon_algorithm")` to `Pref.getBooleanDefaultFalse("external_blukon_algorithm")`